cellprofiler/modules/tram.py

```diff
@@ -0,0 +1,673 @@
+statistic reflects how typical the maximum deviation from smooth time series
+a chosen set of measurements are. Typical fluctuations are determined from
+measurement differences in adjacent time points of objects whose trajectories
+are complete and without splitting events.
+
+<p><b>TODO</b>For an example pipeline using TrAM see the CellProfiler
+<a href="http://www.cellprofiler.org/examples.html#TrAM">Examples</a> webpage.</p>
+
+<h4>Available measurements</h4>
+<ul>
+<li><i>TrAM:</i> The TrAM value for the trajectory. Values near 1 are typical
+for a good trajectory. Large values (typically 3 or higher) are more likely
+to correspond to aberrant tracks. The value <i>nan</i> is assigned to objects
+with partial tracks or those for whom <i>Is_Parent</i> is 1. A histogram of
+all computed TrAM values is displayed to help define a cutoff.</li>
+<li><i>Labels:</i> Each tracked item that has a lineage from the first to the
+last frame is assigned a TrAM label on the last frame. If the final object
+does not arise from a split during tracking, then it has this same unique label
+for its entire track. That label is not assigned to any other objects. But if
+an object arises from a split, then its ancestor object(s) will be assigned
+multiple labels (combined from its progeny). These labels are separated by
+a "|" symbol.</li>
+<li><i>Is_Parent:</i> If the object splits into daughters during its track then 
+flag will be 1, and <i>Labels</i> will contain two or more labels separated by "|".
+Otherwise it is 0 and <i>Labels</i> has only one label.</li>
+<li><i>Split_Trajectory:</i> If the object arose from an ancestor whose trajectory
+split, then this value is 1. Otherwise it is 0.</li>
+</ul>
+<p>
+See Patsch, K <i>et al.</i>, <a href=https://www.nature.com/articles/srep34785>Single cell
+dynamic phenotyping</a>, Scientific Reports 6:34785 (2016)
+""" % (MIN_TRAM_LENGTH)
+
+logger = logging.getLogger(__name__)
+
+class TrAM(cpm.Module):
+    module_name = "TrAM"
+    category = "Object Processing"
+    variable_revision_number = 1
+
+    def create_settings(self):
+        # for them to choose the tracked objects
+        self.object_name = cps.ObjectNameSubscriber(
+                "Tracked objects", cps.NONE, doc="""
+            Select the tracked objects for computing TrAM.""")
+
+        # which measurements will go into the TrAM computation
+        self.tram_measurements = MeasurementMultiChoiceForCategory(
+            "TrAM measurements", category_chooser=self.object_name, doc="""
+            These are measurements for the selected tracked objects which
+            will be used in the TrAM computation. At least one must be selected.
+            Note there may be a delay of a few seconds between the selection of
+            <i>Tracked objects</i> and the update of this selection component.""")
+
+        # Treat X-Y value pairs as Euclidian in the TrAM measure?
+        self.wants_XY_Euclidian = cps.Binary(
```

Comment:
I would consider renaming this property to `self.isotropic`. It'll make more sense to users.